File: server/app/Models/Brand.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use App\Models\Figure;

class Brand extends Model
{
    use HasFactory;

    protected $fillable = [
        "name",
        "image",
        "subImage",
    ];

    public function figures() {
        return $this.hasMany(Figure, "brand", "name");
    }
}

File: client/lib/models/brand.dart
class Brand {
  late int id;
  late String name;
  late String image;
  late String subImage;

  Brand(
      {required this.id,
      required this.name,
      required this.image,
      required this.subImage});

  factory Brand.fromJson(Map<String, dynamic> json) {
    return Brand(
        id: json["id"],
        name: json["name"],
        image: json["image"],
        subImage: json["subImage"]);
  }
}
